packages/slas-react-sdk/src/index.tsx

```diff
@@ -0,0 +1,235 @@
+import React, { createContext, useContext } from "react";
+import { createStore, useStore } from "zustand";
+import { persist } from "zustand/middleware";
+import { helpers, ShopperLogin } from "commerce-sdk-isomorphic";
+import { onClient, getAppOrigin } from "./util";
+import { isExpired } from "./jwt";
+import { MemoryStorage } from "./storage";
+
+enum AuthTypes {
+  REGISTERED = "registered",
+  GUEST = "guest",
+}
+
+interface State {
+  isInitialized: boolean;
+  isInitializing: boolean;
+  isAuthenticated: boolean;
+  authType: AuthTypes;
+  customerId: string;
+  accessToken: string;
+  refreshToken: string;
+  _config: CommerceAPIConfig;
+  login: (username: string, password: string) => void;
+  register: (username: string, password: string) => void;
+  loginAsGuest: () => void;
+  logout: () => void;
+  _initialize: () => void;
+  _refreshAccessToken: () => void;
+  _handleShopperLoginTokenResponse: (res: helpers.TokenResponse) => void;
+}
+
+interface CommerceAPIConfig {
+  clientId: string;
+  organizationId: string;
+  shortCode: string;
+  siteId: string;
+}
+
+interface ProviderProps extends CommerceAPIConfig {
+  proxy?: string;
+  debug?: boolean;
+  children: React.ReactNode;
+}
+
+const ShopperContext = createContext({});
+export const Provider = ({
+  children,
+  organizationId,
+  clientId,
+  shortCode,
+  siteId,
+  proxy,
+  debug,
+}: ProviderProps) => {
+  // TODO: validate config value
+  const config = { organizationId, clientId, siteId, shortCode };
+  const shopperLoginClient = new ShopperLogin({
+    proxy,
+    parameters: config,
+    throwOnBadResponse: true,
+  });
+
+  // we need to store the tokens on server side
+  // for commerce api calls
+  // TODO: test server side
+  const memoryStorage = new MemoryStorage();
+
+  const logger = (text: string) => {
+    if (debug) {
+      console.log(
+        `%c(slas-react-sdk) %c${text}`,
+        "font-weight:bold;",
+        "color:green;"
+      );
+    }
+  };
+
+  const persistStorageOptions = {
+    name: "slas",
+    getStorage: () => {
+      if (onClient()) {
+        return window.localStorage;
+      }
+      return memoryStorage;
+    },
+    partialize: (state: State) => {
+      // Note: unfortunately you can't rename the field here
+      return {
+        accessToken: state.accessToken,
+        refreshToken: state.refreshToken,
+        customerId: state.customerId,
+        authType: state.authType,
+      };
+    },
+  };
+
+  const store = createStore<State>()(
```

Comment:
What benefits do we get over using a simple `useState` hook?

Reply:
I outlined a few benefits in the "Implementation detail" section in the PR description.

One of the benefits we get is syncing state with a storage (localstorage and cookies).